nautobot/docs/development/apps/api/platform-features/prepopulating-data.md

```diff
@@ -0,0 +1,183 @@
+    Relationship.objects.update_or_create(
+        key="location_favorite_animal",
+        defaults={
+            "label": "Location's Favorite Animal",
+            "type": RelationshipTypeChoices.TYPE_ONE_TO_MANY,
+            "source_type": ContentType.objects.get_for_model(Animal),
+            "source_label": "Locations that love this Animal",
+            "destination_type": ContentType.objects.get_for_model(Location),
+            "destination_label": "Favorite Animal",
+        },
+    )
+```
+
+Then, in the `NautobotAppConfig` `ready()` function, we connect this callback function to the `nautobot_database_ready` signal:
+
+```python
+# __init__.py
+
+from nautobot.apps import nautobot_database_ready, NautobotAppConfig
+
+from .signals import create_location_to_animal_relationship
+
+class AnimalSoundsConfig(NautobotAppConfig):
+    # ...
+
+    def ready(self):
+        super().ready()
+        nautobot_database_ready.connect(create_location_to_animal_relationship, sender=self)
+
+config = AnimalSoundsConfig
+```
+
+!!! warning
+It is crucial that you add the `sender=self` parameter to the `connect` method call - otherwise your signal handler will run as many times as the `NautobotAppConfig.ready` method is called, which may be a lot of times.
+
+After writing this code, run `nautobot-server migrate` or `nautobot-server post_upgrade`, then restart the Nautobot server, and you should see that this custom Relationship has now been automatically created.
+
+### Data Migrations
+
+Generally, documentation on this approach can be found in [the corresponding docs section of Django](https://docs.djangoproject.com/en/4.2/topics/migrations/#data-migrations).
+
+There are, however, some specifics to adhere to in the Nautobot ecosystem. For every migration you write in a Nautobot app, you should also implement the `reverse_code` argument of `migrations.RunPython`. This is because the official uninstallation instructions for Nautobot apps require you to first run the migrations for that app in reverse to remove the app's data from the database.
+
+The following is a simple example of a migration that adds the content types from two custom models in your Nautobot app to a default Nautobot status.
+
+```python
+# your_nautobot_app/migrations/0002_custom_status.py
+
+from django.db import migrations
+
+status_content_type_mapping = {
+    "your_nautobot_app.custom_model_a": ["Active", "Decommissioned"],
+    "your_nautobot_app.custom_model_b": ["Active", "Down", "Failed"],
+}
+
+def add_content_types_to_default_statuses(apps, schema_editor):
+    """Adds additional content types to default statuses."""
+    Status = apps.get_model("extras.Status")
+    ContentType = apps.get_model("contenttypes.ContentType")
+    for model, statuses in status_content_type_mapping.items():
+        model_class = apps.get_model(model)
+        for status in statuses:
+            Status.objects.get(name=status).content_types.add(
```

Comment:
Should we use `Status.objects.get_or_create()` here and below as a best practice in case those named statuses don't exist?